lib/webpacker/file_loader.rb

```diff
@@ -4,21 +4,44 @@ class NotFoundError < StandardError; end
   class FileLoaderError < StandardError; end
 
   class_attribute :instance
-  attr_accessor :data
+  attr_accessor :data, :mtime, :path
 
   class << self
-    def load(path = file_path)
+    def load_instance(path = file_path)
+      # Assume production is 100% cached and don't reload if file's mtime not changed
+      cached = self.instance && # if we have a singleton
+        (env == "production" || # skip if production bc always cached
+          (File.exist?(path) && self.instance.mtime == File.mtime(path))) # skip if mtime not changed
+
+      return if cached
```

Comment:
This needs to be extracted and reworked. We shouldn't need this many comments to explain the method. Let's use the Composed Method pattern.